eng/tools/lint-diff/src/runChecks.ts

```diff
@@ -34,63 +35,70 @@ export async function runChecks(
     // If the tags array is empty run the loop once but with a null tag
     const coalescedTags = tags?.length ? tags : [null];
     for (const tag of coalescedTags) {
-      let tagArg = tag ? `--tag=${tag} ` : "";
-
-      let autorestCommand =
-        `npm exec --no -- autorest ` +
-        `--v3 ` +
-        `--spectral ` +
-        `--azure-validator ` +
-        `--semantic-validator=false ` +
-        `--model-validator=false ` +
-        `--message-format=json ` +
-        `--openapi-type=${openApiType} ` +
-        `--openapi-subtype=${openApiSubType} ` +
-        `--use=${dependenciesDir} ` +
-        `${tagArg} ` +
-        `${changedFilePath}`;
-
       console.log(`::group::Autorest for type: ${openApiType} readme: ${readme} tag: ${tag}`);
+
+      const autorestArgs = [
+        "autorest",
+        "--v3",
+        "--spectral",
+        "--azure-validator",
+        "--semantic-validator=false",
+        "--model-validator=false",
+        "--message-format=json",
+        `--openapi-type=${openApiType}`,
+        `--openapi-subtype=${openApiSubType}`,
+        `--use=${dependenciesDir}`,
+      ];
+
+      if (tag) {
+        autorestArgs.push(`--tag=${tag}`);
+      }
+      autorestArgs.push(changedFilePath);
+
+      const autorestCommand = `npm ${autorestArgs.join(" ")}`;
```

Comment:
NIT: Please move this inline to the console.log statement, to emphasize we are **not** executing the joined string, but rather the array or arguments (important to prevent script injection attacks).